### DISCUSSION TITLE
Make a Call with flate file feed

### DISCUSSION THREAD

**Discussion by OliverZemke2021** [abuzuhri/Amazon-SP-API-CSharp#73] [Q&A]
Hi @abuzuhri ,

first at all, your Library i really helpfull, thank you for that.

We have to migrate from the "old" mws to sp-api. We have a call with the feed POST_FLAT_FILE_INVLOADER_DATA, is there in your library an example for making such a call?

Thank you for your help.

**Comment by abuzuhri** [comment]
Hello @OliverZemke2021 

Please check this Example for change inventory 

https://github.com/abuzuhri/Amazon-SP-API-CSharp#feed-submit-for-change-quantity

please let me know if that help you.

**Reply by OliverZemke2021** [reply]
Hi @abuzuhri,

thank you, I will try it with a flat file (csv file with tab separator).
If it will work I will inform you.

Best regards
Olli

**Reply by abuzuhri** [reply]
Please send me sample for that file i will check from my side also

**Reply by OliverZemke2021** [reply]
Good Morning,
is it only possible to create an xml file in your library? The contentype is only xml or json.

**Reply by abuzuhri** [reply]
Hello @OliverZemke2021 
Right now Yes,

Please share the file sample and content Type you want to add it

**Reply by OliverZemke2021** [reply]
Hello @abuzuhri ,

I will need this content type **text/tab-separated-values**

Here is sample file wich I use actual.
[Beispiel_Upload.txt](https://github.com/abuzuhri/Amazon-SP-API-CSharp/files/7903952/Beispiel_Upload.txt)

I hope the file will be ok in the new amazon sp-api

**Reply by abuzuhri** [reply]
Hello @OliverZemke2021 

Please download the source code and try this example , I don't Update the nugget I will wait you to confirm first but depend on my test this will work.

```Csharp
            string text = System.IO.File.ReadAllText(@"C:\Users\tareq\Downloads\Beispiel_Upload.txt");

            var feedresultTXT = amazonConnection.Feed.SubmitFeed(text
                                                    , FeedType.POST_FLAT_FILE_INVLOADER_DATA
                                                    , new List<string>() { MarketPlace.UnitedArabEmirates.ID }
                                                    , null
                                                    , ContentType.TXT);

            Thread.Sleep(1000 * 60);

            var feedOutput = amazonConnection.Feed.GetFeed(feedresultTXT);

            var outPut = amazonConnection.Feed.GetFeedDocument(feedOutput.ResultFeedDocumentId);

            var reportOutpit = outPut.Url;
```

**Comment by abuzuhri** [comment]
```Csharp
string text = System.IO.File.ReadAllText(@"C:\Users\tareq\Downloads\Beispiel_Upload.txt");
var feedresultTXT = amazonConnection.Feed.SubmitFeed(text
                                                    , FeedType.POST_FLAT_FILE_INVLOADER_DATA
                                                    , new List<string>() { MarketPlace.UnitedArabEmirates.ID }
                                                    , null
                                                    , ContentType.TXT);

Thread.Sleep(1000 * 60);

var feedOutput = amazonConnection.Feed.GetFeed(feedresultTXT);

var outPut = amazonConnection.Feed.GetFeedDocument(feedOutput.ResultFeedDocumentId);

var reportOutpit = outPut.Url;
```

**Reply by OliverZemke2021** [reply]
Hello @abuzuhri ,

I have tried to upload. But I got this
{
  "errors": [
    {
      "code": "InvalidInput",
      "message": "Invalid request parameters",
      "details": ""
    }
  ]
}
Do you know wich parameter is invalid?

**Comment by abuzuhri** [comment]
Try this code

```Csharp
string text = System.IO.File.ReadAllText(@"C:\Users\tareq\Downloads\Beispiel_Upload.txt");

            var feedresultTXT = amazonConnection.Feed.SubmitFeed(text
                                                    , FeedType.POST_FLAT_FILE_INVLOADER_DATA
                                                    , new List<string>() { MarketPlace.UnitedArabEmirates.ID }
                                                    , null
                                                    , ContentType.TXT);


            string pathURL = string.Empty;
            while (pathURL == string.Empty)
            {
                Thread.Sleep(1000 * 30);
                var feedOutput = amazonConnection.Feed.GetFeed(feedresultTXT);
                if (feedOutput.ProcessingStatus == AmazonSpApiSDK.Models.Feeds.Feed.ProcessingStatusEnum.DONE)
                {
                    var outPut = amazonConnection.Feed.GetFeedDocument(feedOutput.ResultFeedDocumentId);

                    pathURL = outPut.Url;
                }
            }
```

**Reply by OliverZemke2021** [reply]
Sorry, it is still not working. I got the error in CreateFeed(createFeddSpecification).
Is there somthing missing? Have I forgot to download somting?

This is in my json
{"feedType":"POST_FLAT_FILE_INVLOADER_DATA","marketplaceIds":["A1PA6795UKMFR9"],"inputFeedDocumentId":"amzn1.tortuga.3.53ab5777-17ca-4a2b-a198-c9855c6d64e5.T1TF5QRTC6K9OC"}

var response = ExecuteRequest<CreateFeedResult>();
Here I got the BadRequest

**Reply by abuzuhri** [reply]
contact me by email , the same code work prefect form my side

**Reply by OliverZemke2021** [reply]
You got an email
